src/journal-remote/journal-header-util.c

```diff
@@ -0,0 +1,99 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "escape.h"
+#include "journal-header-util.h"
+#include "string-util.h"
+#include "strv.h"
+
+/* HTTP header name can contains:
+- Alphanumeric characters: a-z, A-Z, and 0-9
+- The following special characters: - and _ */
+#define VALID_HEADER_NAME_CHARS \
+        ALPHANUMERICAL "_-"
+
+#define VALID_HEADER_NAME_LENGTH 40
+
+#define VALID_HEADER_VALUE_CHARS \
+        ALPHANUMERICAL "_ :;.,\\/'\"?!(){}[]@<>=-+*#$&`|~^%"
```

Comment:
Where it is defined? Could you add a reference in the comment?

Also, are there no limit in the length?

Reply:
i haven't found any limits in a standard, but per standard it's allowed to set limits and respond with http 400 in case of too long header names and values
standard allows US-ASCII characters for headers, can i use `is_ascii` there?
this list of symbols I've took from cloudflare website, thought they are relying on a RFC